odxtools/parameters/matchingrequestparameter.py

```diff
@@ -42,11 +43,13 @@ def decode_from_pdu(self, decode_state: DecodeState) -> ParameterValue:
         if self.byte_position is not None:
             decode_state.cursor_byte_position = decode_state.origin_byte_position + self.byte_position
 
-        byte_position = decode_state.cursor_byte_position
-        bit_position = self.bit_position or 0
-        byte_length = (8 * self.byte_length + bit_position + 7) // 8
-        result = decode_state.coded_message[byte_position:byte_position + byte_length]
-        decode_state.cursor_byte_position += byte_length
+        result, decode_state.cursor_byte_position = DiagCodedType._extract_atomic_value(
```

Comment:
Is it just me, or a function that takes `decode_state` and `self` would be easier to understand and re-use?

Reply:
this is the lowest level method which IMO should be usable without a `decode_state` and even without a DiagCodedType object. (also, DCTs already have a `decode_from_pdu()` method...)

Reply:
then why it is in the DiagCodedType class?

Reply:
because usually (all occasions instead of this one), it is called from there.

Reply:
(`_extract_atomic_value()` deals with retrieving a non-byte-aligned object from a blob of binary data, while `decode_from_pdu()` also deals with the "book keeping" of updating `decode_state`.)

Reply:
maybe put the function in the decode state itself?